### DISCUSSION TITLE
How can store graph index

### DISCUSSION THREAD

**Discussion by ragesh2000** [run-llama/llama_index#14954] [Q&A]
Iam referring to [https://github.com/run-llama/llama_index/blob/main/docs/docs/examples/cookbooks/GraphRAG_v1.ipynb](https://github.com/run-llama/llama_index/blob/main/docs/docs/examples/cookbooks/GraphRAG_v1.ipynb) for creating GraphRAG using llamaindex. So once an index is created how can we store it ? I have tried the conventional `.persist` method but got an error.

```
index = PropertyGraphIndex(
    nodes=nodes,
    property_graph_store=GraphRAGStore(),
    kg_extractors=[kg_extractor],
    show_progress=True,
)
index.property_graph_store.build_communities()
index.storage_context.persist(persist_dir="./graphstore2")
storage_context = StorageContext.from_defaults(persist_dir="./graphstore2")
loaded_index = load_index_from_storage(storage_context)

query_engine = GraphRAGQueryEngine(
    graph_store=loaded_index.property_graph_store, llm=llm
)
response = query_engine.query(
    "What are the main news discussed in the document?"
)
```
Error
`instance of GraphRAGStore expected (type=type_error.arbitrary_type; expected_arbitrary_type=GraphRAGStore)`

How can I store the index and load later?

**Comment by KYLEEZTSMC** [comment]
I dig into llama_index [source code](https://github.com/run-llama/llama_index/blob/main/llama-index-core/llama_index/core/storage/storage_context.py) and found out this:
`
storage_context = StorageContext.from_defaults(persist_dir="./storage", property_graph_store=GraphRAGStore())

test_index = load_index_from_storage(storage_context)

test_query_engine = GraphRAGQueryEngine(
    graph_store=test_index.property_graph_store, 
    llm=llm,
    index=test_index,
)
`
**property_graph_store** parameter can be used to solve your problem.
